packages/core/data-transfer/src/engine/validation/provider.ts

```diff
@@ -0,0 +1,27 @@
+import { capitalize } from 'lodash/fp';
+
+import type { IDestinationProvider, ISourceProvider, ProviderType } from '../../../types';
+import { TransferEngineValidationError } from '../errors';
+
+const reject = (reason: string): never => {
+  throw new TransferEngineValidationError(`Invalid provider supplied. ${reason}`);
```

Comment:
Can we make `TransferEngineValidationError` accept an object so that `reason` can be received at a higher level? (just a minor thing I'm fine with adding to our future TODO list)